### PR TITLE
Interact with pipeline triggers

### PR DESCRIPTION
**Pull request by christophermaier**

### CODE REVIEW

File: lib/cog_api/fake/client.ex [hunk 1]
```diff
@@ -172,6 +173,24 @@ defmodule CogApi.Fake.Client do
     Rules.delete(rule_id, endpoint)
   end
 
+  def trigger_by_name(%Endpoint{}=endpoint, name),
```
Comment (drapergeek) [inline lib/cog_api/fake/client.ex]: What do you think about using the `do/end` here just to be consistent with the rest of the file?

Comment (christophermaier) [inline lib/cog_api/fake/client.ex]: Done

File: lib/cog_api/client.ex [hunk 2]
```diff
@@ -62,6 +63,13 @@ defmodule CogApi.Client do
   @callback rule_delete(String.t, %Endpoint{}) :: atom
   @callback rule_delete(String.t, %Endpoint{}) :: {atom, [String.t]}
 
+  @callback trigger_by_name(%Endpoint{}, String.t) :: {:ok, %Trigger{}} | {:error, term}
```
Comment (drapergeek) [inline lib/cog_api/client.ex]: Should this be `trigger_show_by_name` just to make it clear that this is the `show` option?

**Comment by drapergeek** [comment]
One comment, then :shipit:

File: lib/cog_api/client.ex (hunk 2 above)
Comment (christophermaier) [inline lib/cog_api/client.ex]: Not a bad idea... quick change coming up

**Pull request merged**